Cython/Utility/ModuleSetupCode.c

```diff
@@ -351,21 +355,35 @@
     #define CYTHON_USE_ASYNC_SLOTS 1
   #endif
   #ifndef CYTHON_USE_PYLONG_INTERNALS
-    #define CYTHON_USE_PYLONG_INTERNALS 1
+    #ifdef Py_GIL_DISABLED
+      #define CYTHON_USE_PYLONG_INTERNALS 0
+    #else
+      #define CYTHON_USE_PYLONG_INTERNALS 1
+    #endif
   #endif
-  #ifndef CYTHON_USE_PYLIST_INTERNALS
-    #define CYTHON_USE_PYLIST_INTERNALS 1
+  #ifdef Py_GIL_DISABLED
+    #undef CYTHON_USE_PYLIST_INTERNALS
+    #define CYTHON_USE_PYLIST_INTERNALS 0
+  #else
+    #ifndef CYTHON_USE_PYLIST_INTERNALS
+      #define CYTHON_USE_PYLIST_INTERNALS 1
+    #endif
   #endif
```

Comment:
Let's just chain these cases like this:
```suggestion
  #ifdef Py_GIL_DISABLED
    #undef CYTHON_USE_PYLIST_INTERNALS
    #define CYTHON_USE_PYLIST_INTERNALS 0
  #elif !defined(CYTHON_USE_PYLIST_INTERNALS)
    #define CYTHON_USE_PYLIST_INTERNALS 1
  #endif
```